api/v1alpha1/datadogagent_types.go

```diff
@@ -877,6 +883,8 @@ type ExternalMetricsConfig struct {
 	// empty.
 	// +optional
 	Endpoint *string `json:"endpoint,omitempty"`
+
```

Comment:
could you add a comment that explains how this credentials behave compare to the "principal" credential. (example priority)